esx_service/vmdk_ops.py

```diff
@@ -1670,22 +1683,39 @@ def set_vol_opts(name, tenant_name, options):
 
     return False
 
+def wait_ops_in_flight():
+    global STOP_RETRY_COUNT
+
+    # Set the STOP flag to true
+    stopEvent.set()
+
+    # Wait until the in-flight ops counter goes down to zero
+    while ((opsCounter.value > 0) and (STOP_RETRY_COUNT > 0)):
+        logging.info("Service is being stopped. Waiting for completion of %d operations in flight...", opsCounter.value)
+        time.sleep(STOP_RETRY_SLEEP)
+        STOP_RETRY_COUNT -= 1
+
+    if (opsCounter.value == 0):
+        logging.info("All in-flight operations are completed. Exiting...")
+    else:
+        logging.info("In-flight operations are taking too long to complete")
+
 def msg_about_signal(signalnum, details="exiting"):
     logging.warn("Received signal num: %d, %s.", signalnum, details)
```

Comment:
Log should also mention that new requests are stopped and waiting for in-flight commands to drain.

Reply:
We have those logs inside wait_ops_in_flight().